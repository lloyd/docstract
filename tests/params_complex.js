// a function which accepts arguments embedded inside an object

/**
 * @function foo
 *
 * This function ... blah blah blah
 *
 * @param options {object
 *
 *   @prop url {string}
 *   This is the url to which the request will be made.
 *
 *   @prop [onComplete] {function}
 *
 *   This function will be called when the request has received a response (or in
 *   terms of XHR, when `readyState == 4`). The function is passed a `Response`
 *   object.
 *
 *   @prop [headers] {object}
 *   An unordered collection of name/value pairs representing headers to send
 *   with the request.
 *
 *   @prop [content] {string,object}
 *
 *   The content to send to the server. If `content` is a string, it
 *   should be URL-encoded (use `encodeURIComponent`). If `content` is
 *   an object, it should be a collection of name/value pairs. Nested
 *   objects & arrays should encode safely.
 *
 *   For `GET` requests, the query string (`content`) will be appended
 *   to the URL. For `POST` requests, the query string will be sent as
 *   the body of the request.
 *
 *   @prop [contentType] {string}
 *
 *   The type of content to send to the server. This explicitly sets the
 *   `Content-Type` header. The default value is `application/x-www-form-urlencoded`.
 * }
 */
