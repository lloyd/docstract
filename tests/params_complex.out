{
  "filename": "tests/params_complex.js", 
  "functions": [
    {
      "desc": "This function ... blah blah blah", 
      "name": "foo", 
      "params": [
        {
          "name": "options", 
          "type": {
            "name": "object", 
            "properties": [
              {
                "desc": "  This is the url to which the request will be made.", 
                "name": "url", 
                "source_lines": [
                  3, 
                  39
                ], 
                "type": "string"
              }, 
              {
                "desc": "  This function will be called when the request has received a response (or in\n  terms of XHR, when `readyState == 4`). The function is passed a `Response`\n  object.", 
                "name": "onComplete", 
                "optional": true, 
                "source_lines": [
                  3, 
                  39
                ], 
                "type": "function"
              }, 
              {
                "desc": "  An unordered collection of name/value pairs representing headers to send\n  with the request.", 
                "name": "headers", 
                "optional": true, 
                "source_lines": [
                  3, 
                  39
                ], 
                "type": "object"
              }, 
              {
                "desc": "  The content to send to the server. If `content` is a string, it\n  should be URL-encoded (use `encodeURIComponent`). If `content` is\n  an object, it should be a collection of name/value pairs. Nested\n  objects & arrays should encode safely.\n\n  For `GET` requests, the query string (`content`) will be appended\n  to the URL. For `POST` requests, the query string will be sent as\n  the body of the request.", 
                "name": "content", 
                "optional": true, 
                "source_lines": [
                  3, 
                  39
                ], 
                "type": "string,object"
              }, 
              {
                "desc": "  The type of content to send to the server. This explicitly sets the\n  `Content-Type` header. The default value is `application/x-www-form-urlencoded`.", 
                "name": "contentType", 
                "optional": true, 
                "source_lines": [
                  3, 
                  39
                ], 
                "type": "string"
              }
            ]
          }
        }
      ], 
      "source_lines": [
        3, 
        39
      ]
    }
  ], 
  "module": "params_complex"
}
